app/helpers/hosts_helper.rb

```diff
@@ -245,44 +245,6 @@ def name_field(host)
     (SETTINGS[:unattended] and host.managed?) ? host.shortname : host.name
   end
 
-  def show_templates
-    unless SETTINGS[:unattended] and @host.managed?
-      return alert(:class=> 'alert-warning',
-                   :text => _("Provisioning support is disabled or this host is not managed"))
-    end
-    begin
-      templates = Hash[TemplateKind.order(:name).map do |k|
-        template = @host.provisioning_template(:kind => k.name)
-        next if template.nil?
-        [k.name, template]
-      end.compact]
-    rescue => e
-      return case e.to_s
-      when "Must provide an operating systems"
-        _("Unable to find templates as this host has no operating system")
-      else
-        e.to_s
-      end
-    end
-
-    return _("No template found") if templates.empty?
```

Comment:
I think you are missing this line in the refactoring - no templates will lead to an empty table instead of a text explanation.